### PR TITLE
feat: Show Warning for Unconfigured Time Machine

### PR DESCRIPTION
**Pull request by jordanleven**
# Description
Currently, the policy check scripts works by retrieving a list of all volumes registered as backups with Time Machine and individually checking their encryption status. However, in the event a user has _not_ registered any volumes, the script doesn't clearly display whether or not the check has passed or failed. In this branch, we'll update the output to warn the user's who don't use Time Machine that the policy check (specifically for Time Machine) hasn't run.

## Validation for those using Time Machine
1. Run `curl -s https://raw.githubusercontent.com/sparkbox/standard/fix--no-registered-tm-backups/security/security_policy_compliance/policy-checks.sh | sh`
- [x]  Confirm backup checks run as expected.

## Validation for those not using Time Machine
1. Run `curl -s https://raw.githubusercontent.com/sparkbox/standard/fix--no-registered-tm-backups/security/security_policy_compliance/policy-checks.sh | sh`
- [ ]  Confirm a warning is displayed indicating that encryption checks did not run.

### CODE REVIEW

File: security/security_policy_compliance/policy-checks.sh [hunk 1]
```diff
@@ -22,6 +22,12 @@ function success {
   echo "✅ \033[1;32m${DETAILS}\033[0m\n";
 }
 
+function warning {
+  DETAILS=$1
+
+  echo "⚠️ \033[1;33m ${DETAILS}\033[0m\n";
```
Comment (cromwellryan) [inline security/security_policy_compliance/policy-checks.sh:28]: Perfect emoji choice ⚠️ 👍

File: security/security_policy_compliance/policy-checks.sh [hunk 2]
```diff
@@ -38,6 +44,10 @@ start "Checking Time Machine Backup Disk Encryption"
 # Replace spaces with three underscores so we can use spaces as a delimiter in an array
 TIME_MACHINE_VOLUMES=$(tmutil destinationinfo | grep -e "Name" | sed 's/Name.*: //' | sed -e "s/ /___/g")
 
+if [ "${TIME_MACHINE_VOLUMES}" == "" ]; then
+  warning "No backup disks configured with Time Machine."
```
Comment (cromwellryan) [inline security/security_policy_compliance/policy-checks.sh:48]: Can we give this URL as help similar to the `fail` outcome?

https://github.com/sparkbox/standard/blob/1ca790bf2e5d553264b5d746ab3a9c91a6ffb766/security/security_policy_compliance/timemachine.md#first-time-setup

File: security/security_policy_compliance/policy-checks.sh [hunk 3]
```diff
@@ -56,10 +66,10 @@ for VOLUME in $TIME_MACHINE_VOLUMES; do
 
   case $VOLUME_ENCRYPTED in
     Yes)
-      success "${VOLUME_NAME}"
+      success "\"${VOLUME_NAME}\" is encrypted."
```
Comment (cromwellryan) [inline security/security_policy_compliance/policy-checks.sh:69]: Good addition 🙌

**Review by cromwellryan — COMMENTED** [review]
Still needs checked on a device without backups.

File: security/security_policy_compliance/policy-checks.sh (hunk 2 above)
Comment (jordanleven) [reply security/security_policy_compliance/policy-checks.sh:48]: Ah, good catch! I just updated this in aa8b0f7eee3d77a925a203ae1ce7dec8854e514d. How's it looking now?

File: security/security_policy_compliance/policy-checks.sh (hunk 3 above)
Comment (jordanleven) [reply security/security_policy_compliance/policy-checks.sh:69]: Thanks! 😄

File: security/security_policy_compliance/policy-checks.sh (hunk 1 above)
Comment (jordanleven) [reply security/security_policy_compliance/policy-checks.sh:28]: Gracias! 😄

File: security/security_policy_compliance/policy-checks.sh (hunk 2 above)
Comment (jordanleven) [reply security/security_policy_compliance/policy-checks.sh]: Ah, good catch! I updated it in fea2d79f5031a8d45bd5a5469f68f947d8dc8d13. How's it looking now?

**Comment by changangus** [comment]
Tested on a machine not using Time Machine:
Screen Shot 2021-07-16 at 9 17 05 AM

**Review by changangus — APPROVED** [review]
Tested on a machine without Time Machine: 
Screen Shot 2021-07-16 at 9 17 05 AM

**Comment by alicearussell** [comment]
Screenshot running on macOS 11.4 with no Time Machine enabled.

Screen Shot 2021-07-16 at 8 15 46 AM

**Review by alicearussell — APPROVED** [review]

**Pull request closed without merging**